jujugui/static/gui/src/app/app.js

```diff
@@ -982,11 +982,15 @@ YUI.add('juju-gui', function(Y) {
     */
     _renderLogin: function(err) {
       document.getElementById('loading-message').style.display = 'none';
+      const controllerAPI = this.controllerAPI;
+      const loginToController = controllerAPI.loginWithMacaroon.bind(
+        controllerAPI, this.bakeryFactory.get('juju'));
       ReactDOM.render(
         <window.juju.components.Login
           setCredentials={this.env.setCredentials.bind(this.env)}
           isLegacyJuju={this.isLegacyJuju()}
           loginToAPIs={this.loginToAPIs.bind(this)}
+          loginToController={loginToController}
```

Comment:
Can you elaborate on why you're adding another method here? By doing this and not using the `loginToAPIs` method then you're losing all of the callback handling as defined in the `_apiLoginHandler` in app.js